recipes/qt-advanced-docking-system/all/conanfile.py

```diff
@@ -26,70 +33,79 @@ class QtADS(ConanFile):
 
-    def config_options(self):
-        if self.settings.os == "Windows":
-            del self.options.fPIC
+    @property
+    def _min_cppstd(self):
+        if self._qt_major >= 6:
+            return 17
+        else:
+            return 14
 
-    def configure(self):
-        if self.options.shared:
-            del self.options.fPIC
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires(f"qt/{self._qt_version}")
+        self.requires("qt/[>=6.0 <7]", transitive_headers=True, transitive_libs=True, run=can_run(self))
+        self.requires("libpng/[>=1.6 <2]")
 
-    def source(self):
-        get(self, **self.conan_data["sources"][self.version], strip_root=True,
-                  destination=self._source_subfolder)
+    def validate(self):
+        check_min_cppstd(self, self._min_cppstd)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
+    def build_requirements(self):
+        if not can_run(self):
+            self.tool_requires("qt/<host_version>", options={"gui": False, "widgets": False})
 
-        self._cmake = CMake(self)
-        self._cmake.definitions["ADS_VERSION"] = self.version
-        self._cmake.definitions["BUILD_EXAMPLES"] = "OFF"
-        self._cmake.definitions["BUILD_STATIC"] = not self.options.shared
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-        self._cmake.configure()
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["ADS_VERSION"] = self.version
+        tc.variables["BUILD_EXAMPLES"] = "OFF"
+        tc.variables["BUILD_STATIC"] = not self.options.shared
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def _patch_sources(self):
-        apply_conandata_patches(self)
-
-        replace_in_file(self,
-            f"{self.source_folder}/{self._source_subfolder}/src/ads_globals.cpp",
-            "#include <qpa/qplatformnativeinterface.h>",
-            f"#include <{self._qt_version}/QtGui/qpa/qplatformnativeinterface.h>"
+        qt_version = self.dependencies["qt"].ref.version
+        replace_in_file(self, os.path.join(self.source_folder, "src", "ads_globals.cpp"),
+                        "#include <qpa/qplatformnativeinterface.h>",
+                        f"#include <{qt_version}/QtGui/qpa/qplatformnativeinterface.h>",
         )
 
     def build(self):
         self._patch_sources()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
         cmake.install()
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
```

Comment:
```suggestion
        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
        copy(self, "gnu-lgpl-v2.1.md", self.source_folder, os.path.join(self.package_folder, "licenses"))
```

The upstream copies gnu-lgpl-2.1.md file as well when running install command. I would suggest copying it to the package too, just in case.